spopt/locate/flow.py

```diff
@@ -0,0 +1,2630 @@
+        Dict
+            A dictionary containing detailed solver information
+        """
+        if not self.solver_type:
+            raise ValueError("No solver results available. Call solve() first.")
+
+        solver_details = {
+            "solver_type": self.solver_type,
+            "solver_status": self.status,
+        }
+
+        if self.solver_type == "greedy":
+            solver_details.update({
+                "total_iterations": len(self.greedy_iterations),
+                "iteration_details": self.greedy_iterations,
+                "marginal_contributions": dict(
+                    sorted(
+                        self.marginal_contributions.items(), 
+                        key=lambda x: x[1], 
+                        reverse=True
+                    )[:10]  
+                )
+            })
+        else:  
+            solver_details.update({
+                "pulp_status": self.pulp_status,
+                "model_name": self.model.name if self.model else 'N/A',
+                "solver_stats": self.solver_stats,
+                "lagrange_multipliers": dict(list(self.lagrange_multipliers.items())[:5]),
+                "reduced_costs": dict(list(self.reduced_costs.items())[:5])
+            })
+
+        return solver_details
+
+    @property
+    def selected_facilities(self):
+        """Read facility selection from PuLP variables."""
+        if not self.facility_vars:
+            return {}
+        
+        facilities = {}
+        for k, var in enumerate(self.facility_vars):
+            if var.varValue and var.varValue > 0.5:
+                site = self.candidate_sites[k]
+                if self.capacity is not None:
+                    facilities[site] = int(round(var.varValue))
+                else:
+                    facilities[site] = 1
+        return facilities
+
+    def write_csv(self, filename: str, include_iterations: bool = True) -> None:
+        """
+        Export solution to CSV files.
+        """
+        if not self.selected_facilities:
+            raise ValueError("No solution available. Call solve() first.")
+        
+        if not hasattr(self, 'flow_coverage'):
+            self.get_flow_coverage()
+
+        facilities_df = pd.DataFrame(
+            [{"facility_id": k, "modules": v} for k, v in self.selected_facilities.items()]
+        )
+        facilities_df.to_csv(filename.replace(".csv", "_facilities.csv"), index=False)
+
+        coverage_data = []
+        for od_pair, coverage in self.flow_coverage.items():
+            coverage_data.append({
+                "origin": od_pair[0],
+                "destination": od_pair[1],
+                "flow_volume": coverage["flow_volume"],
+                "covered_proportion": coverage["covered_proportion"],
+                "covered_volume": coverage["covered_volume"],
+            })
+        coverage_df = pd.DataFrame(coverage_data)
+        coverage_df.to_csv(filename.replace(".csv", "_coverage.csv"), index=False)
```

Comment:
We can let the user decide which format of document they would like, following pandas DataFrame specification. Then we can just return a dataframe, and the user can convert to a document.